src/app/pages/modes.component.ts

```diff
@@ -43,6 +51,43 @@ class ModesComponent extends HTMLElement {
 		this.selectModeBtn?.removeEventListener('click', () => {
 		});
 	}
+
+	attributeChangedCallback(name: string, oldValue: string, newValue: string): void {
+		if ('data-list-mode' === name) {
+			this.displayListMode(JSON.parse(newValue));
+		}
+	}
+
+	displayListMode = (json: any): void => {
+		const listMode = json.modes;
+		const selectedMode = json.selectedMode;
+		let radioModeList = '';
+		listMode.forEach((mode: any) => {
+			let radioMode = `<app-select-mode data-label="${Object.entries(mode)[0][0]}"></app-select-mode>`;
+			radioModeList = radioModeList + radioMode;
+		});
+		this.selectModeZone!.innerHTML = radioModeList;
+
+		this.selectModeZone.querySelectorAll('app-select-mode').forEach((selectMode) => {
+			let element = selectMode as HTMLElement;
+			if (element.dataset.label === selectedMode) {
+				selectMode.querySelector('input').checked = true;
+			} else {
+				selectMode.querySelector('input').checked = false;
+			}
+		});
+	}
+
+	getSelectedMode = (): string => {
+		let selectedMode = '';
+		let inputs = this.querySelectorAll<HTMLInputElement>('input[name="modes"]');
+		inputs.forEach((input: HTMLInputElement) => {
+			if (input.checked) {
+				selectedMode = input.value;
+			}
+		});
+		return selectedMode;
```

Comment:
Ici, si on ne peut en avoir qu’un coché à la fois, plutôt que de boucler sur tous les éléments tu pourrais faire un `querySelector('input:checked')` directement et récupérer sa valeur : la méthode entière deviendrait suelement ça :

```suggestion
		return this.querySelector('input:checked').value;
```